File: backend/schemas/environment.py
### Environment schema ###

def environment_schema(environment) -> dict:
    return {"id": str(environment["_id"]), # Pasamos Objeto a String
            "company_id": environment["company_id"],
            "city_id": environment["city_id"],
            "type_id": environment["type_id"],
            "name": environment["name"],
            "address": environment["address"],
            "gps_location": environment["gps_location"],
            "description": environment["description"]
            }


def environments_schema(environments) -> list:
    return [environment_schema(environment) for environment in environments]

File: frontend/src/types/environment/environment.ts
import { z } from "zod";


export const EnvironmetSchema = z.object({
    id: z.string(),
    company_id: z.string(),
    city_id: z.string(),
    type_id: z.string(),
    name: z.string(),
    address: z.string(),
    gps_location: z.string(),
    description: z.string(),
})


export type Environment = z.infer<typeof EnvironmetSchema>

export type CreateEnvironment = Pick<Environment, 'company_id' | 'city_id' |
    'type_id' | 'name' | 'address' | 'gps_location' | 'description'>

// Listar
export const EnvironmentsListSchema = z.array(EnvironmetSchema)
